### PR TITLE
Add Go metrics

### PR DESCRIPTION
**Pull request by pellared**
Towards https://github.com/open-telemetry/opentelemetry-go/issues/3186

Based on https://opentelemetry.io/docs/instrumentation/js/manual/#metric. I did my best to make the examples a little more meaningful.

Preview: https://deploy-preview-3235--opentelemetry.netlify.app/docs/instrumentation/go/manual/#metrics

## Next PR

I want to create a separate PR for Views as I will probably document it differently (simpler as most of the docs is already in place here: https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric). I plan to start working on it as soon as this PR is merged.

## Docs PR Checklist

- [x] This PR is for a documentation page whose authoritative copy is in the
      opentelemetry.io repository.

### CODE REVIEW

File: content/en/docs/instrumentation/go/manual.md
Comment (pellared) [inline content/en/docs/instrumentation/go/manual.md:495]: Stop reviewing here 😉

File: content/en/docs/instrumentation/go/manual.md
Comment (cartermp) [inline content/en/docs/instrumentation/go/manual.md:344]:
```suggestion
If a `MeterProvider` is not created, the OpenTelemetry APIs for metrics will use
a no-op implementation and fail to generate data. Therefore, you have modify the
source code to include the SDK initialization code using the [`go.opentelemetry.io/otel`][],
[`go.opentelemetry.io/otel/sdk/metric`][], [`go.opentelemetry.io/otel/sdk/resource`][], and
[`go.opentelemetry.io/otel/exporters/stdout/stdoutmetric`][] packages.
```

Comment (cartermp) [reply content/en/docs/instrumentation/go/manual.md:344]: Recommend pulling these out into bullet points, actually.

File: content/en/docs/instrumentation/go/manual.md
Comment (cartermp) [inline content/en/docs/instrumentation/go/manual.md:429]: In the tracing manual instrumentation we don't do this (using just the API to acquire the tracer). I wrote those docs ages ago, though, and they were approved then but it's maybe worth re-addressing in a separate PR. What do you think? I'll submit that PR separately.

Comment (pellared) [reply content/en/docs/instrumentation/go/manual.md:429]: In "Getting a Tracer" section we probably suggest doing to much and it should be splitted into something like "Initialize Tracing" and "Aquire a Tracer". For sure it should be in a seperate PR. I already seen some things in "tracing" sections that should be probably changed/improved but I want this PR to only focus on translating the "metrics" from JS to Go. Then I also want to improve the "Getting Started" for Go. After these two PRs are merged, I think it would be easier to consolidate Tracing and Metrics docs. Maybe we could even change the code snippets so that they refer somehow to the "Roll Dice" service?

File: content/en/docs/instrumentation/go/manual.md
Comment (pellared) [reply content/en/docs/instrumentation/go/manual.md]: https://github.com/open-telemetry/opentelemetry.io/pull/3235/commits/b977a5d68b51cd7e21c55306deaacc862d4b7997

**Comment by pellared** [comment]
@open-telemetry/go-approvers Let me know if you like this examples and you feel that they should be included in https://pkg.go.dev/go.opentelemetry.io/otel/metric.

@open-telemetry/docs-approvers If we do so then do you think it would make sense to simply refer to the examples e.g. like this: https://pkg.go.dev/go.opentelemetry.io/otel/metric#example-Meter-Synchronous?

File: content/en/docs/instrumentation/go/manual.md
Comment (MrAlias) [inline content/en/docs/instrumentation/go/manual.md:454]:
```suggestion
  the current executing thread to wait for the measurement
```

File: content/en/docs/instrumentation/go/manual.md
Comment (MrAlias) [inline content/en/docs/instrumentation/go/manual.md:457]:
```suggestion
- There is no known timestamp for a measurement value
```

File: content/en/docs/instrumentation/go/manual.md
Comment (MrAlias) [inline content/en/docs/instrumentation/go/manual.md:465]:
```suggestion
Counters can be used to measure a non-negative, increasing value.
```

File: content/en/docs/instrumentation/go/manual.md
Comment (MrAlias) [inline content/en/docs/instrumentation/go/manual.md:467]: nit

```suggestion
For example, here's how you might report several calls for an HTTP handler:
```

File: content/en/docs/instrumentation/go/manual.md
Comment (MrAlias) [inline content/en/docs/instrumentation/go/manual.md:498]:
```suggestion
For example, here's how you might report several items of some collection:
```

File: content/en/docs/instrumentation/go/manual.md
Comment (MrAlias) [inline content/en/docs/instrumentation/go/manual.md:607]:
```suggestion
```go
```

File: content/en/docs/instrumentation/go/manual.md
Comment (MrAlias) [inline content/en/docs/instrumentation/go/manual.md:615]:
```suggestion
// registerDBMetrics registers asynchronous metrics for the provided db.
```

File: content/en/docs/instrumentation/go/manual.md
Comment (pellared) [reply content/en/docs/instrumentation/go/manual.md]: https://github.com/open-telemetry/opentelemetry.io/pull/3235/commits/df79ee2a0cb5a74dda9167a9adfd55e7ece7c34e

File: content/en/docs/instrumentation/go/manual.md
Comment (pellared) [reply content/en/docs/instrumentation/go/manual.md]: https://github.com/open-telemetry/opentelemetry.io/pull/3235/commits/df79ee2a0cb5a74dda9167a9adfd55e7ece7c34e

File: content/en/docs/instrumentation/go/manual.md
Comment (pellared) [reply content/en/docs/instrumentation/go/manual.md:301]: `an` is in previous line.

File: content/en/docs/instrumentation/go/manual.md
Comment (pellared) [reply content/en/docs/instrumentation/go/manual.md:302]: I think that your proposal adds passive voice. But I may be wrong.

File: content/en/docs/instrumentation/go/manual.md
Comment (pellared) [reply content/en/docs/instrumentation/go/manual.md]: https://github.com/open-telemetry/opentelemetry.io/pull/3235/commits/e78fbbabfec8e7ae882663ce02a1604b101aa3f1

File: content/en/docs/instrumentation/go/manual.md
Comment (pellared) [reply content/en/docs/instrumentation/go/manual.md]: https://github.com/open-telemetry/opentelemetry.io/pull/3235/commits/e78fbbabfec8e7ae882663ce02a1604b101aa3f1

File: content/en/docs/instrumentation/go/manual.md
Comment (pellared) [reply content/en/docs/instrumentation/go/manual.md]: https://github.com/open-telemetry/opentelemetry.io/pull/3235/commits/e78fbbabfec8e7ae882663ce02a1604b101aa3f1

**Comment by pellared** [comment]
@MrAlias thanks for your review. Most of your suggestions have been applied.

File: content/en/docs/instrumentation/go/manual.md
Comment (dmathieu) [inline content/en/docs/instrumentation/go/manual.md:422]: This example only uses stdout, which works but is limited. How about linking to the prometheus (and once stable and added, the otel-collector) examples in the Go repository?

File: content/en/docs/instrumentation/go/manual.md
Comment (dmathieu) [inline content/en/docs/instrumentation/go/manual.md:479]:
```suggestion
	apiCounter, err := meter.Int64Counter(
```

File: content/en/docs/instrumentation/go/manual.md
Comment (dmathieu) [inline content/en/docs/instrumentation/go/manual.md:691]:
```suggestion
You can add Attributes by using the
```

File: content/en/docs/instrumentation/go/manual.md
Comment (dmathieu) [inline content/en/docs/instrumentation/go/manual.md:695]:
```suggestion
options.
```

File: content/en/docs/instrumentation/go/manual.md
Comment (pellared) [reply content/en/docs/instrumentation/go/manual.md:422]: This is for sake of consistency with JavaScript docs. I also noticed that other "getting started" docs also start with console exporters.

**Review by dmathieu — APPROVED** [review]

**Review by cartermp — APPROVED** [review]

**Comment by pellared** [comment]
> @open-telemetry/go-approvers Let me know if you like this examples and you feel that they should be included in https://pkg.go.dev/go.opentelemetry.io/otel/metric docs.

I had a chat with @MrAlias and he agreed that we should try to have the same examples in https://pkg.go.dev/go.opentelemetry.io/otel/metric and https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric

> @open-telemetry/docs-approvers If we do so then do you think it would make sense to simply refer to the examples e.g. like this: https://pkg.go.dev/go.opentelemetry.io/otel/metric#example-Meter-Synchronous?

I think it will be better to have the examples here as well. It will be more readable. Still we could add hyperlinks to https://pkg.go.dev when add the examples there.

**Comment by cartermp** [comment]
🎉

**Pull request merged**
